cmd/repo-query/repo-query.go

```diff
@@ -0,0 +1,353 @@
+package repoquery
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const repoQueryURL string = "https://opensauced.tools"
+
+type Options struct {
+	// URL is the git repo URL that will be indexed
+	URL string
+
+	branch string
+}
+
+const repoQueryLongDesc string = `WARNING: Proof of concept feature.
+
+The repo-query command takes a URL to a git repository and indexes it so
+users can ask questions about it.`
+
+func NewRepoQueryCommand() *cobra.Command {
+	opts := &Options{}
+
+	cmd := &cobra.Command{
+		Use:   "repo-query url [flags]",
+		Short: "Ask questions about a git repository",
+		Long:  repoQueryLongDesc,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 1 {
+				return errors.New("only a single url can be ingested at a time")
+			}
+			if len(args) == 0 {
+				return errors.New("must specify the URL of a git repository")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.URL = args[0]
+			return run(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.branch, "branch", "b", "HEAD", "The branch to index")
+
+	return cmd
+}
+
+func getOwnerAndRepo(url string) (owner, repo string) {
+	// Remove the "https://github.com/" prefix from the URL
+	url = strings.TrimPrefix(url, "https://github.com/")
+
+	// Split the remaining URL path into segments
+	segments := strings.Split(url, "/")
+
+	// The first segment is the owner, and the second segment is the repository name
+	if len(segments) >= 2 {
+		owner = segments[0]
+		repo = segments[1]
+	} else {
+		// URL is not in the expected format
+		owner = ""
+		repo = ""
+	}
```

Comment:
Is this case at all valid? This would make sense to return an error here that can be handled by the caller.